packages/eslint-plugin/lib/rules/toggle.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * @fileoverview Opticks
+ * @author Jop
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: "problem",
+    docs: {
+      description: "Detects stale code from expired Opticks experiments",
+      recommended: false,
+      url: null, // URL to the documentation page for this rule
+    },
+    fixable: "code",
+    hasSuggestions: true,
+    schema: [], // Add a schema if the rule has options
+    messages: {
+      ExperimentNotConfigured:
+        "Looks like this experiment is not configured. Please make sure the experiment is added to the experiments config file.",
+      ExperimentConcluded:
+        "Looks like this experiment concluded, and can be cleaned up. The winning variant is {{winningVariant}}.",
+      AddWinningVariant:
+        "If the experiment is concluded, add the winning variant.",
+      VariableAssignment:
+        "It is okay to assign the result of a toggle to a variable, but you might be better off calling the toggle inline for automatic clean up.",
+      InvalidNrOfVariants:
+        "Invalid number of variants. Toggles require either 0, 2, or more variants.",
+      AddNullBVariant:
+        "If the b side is not supposed to do anything, add a null value.",
+    },
+  },
+
+  create(context) {
+    // QUESTION it seems the test runner reads from `context.settings`
+    // while .eslintrc.js reads from `settings`
+    const settings = context.settings || settings;
+    const { opticks } = settings;
+    // variables should be defined here
+
+    //----------------------------------------------------------------------
+    // Helpers
+    //----------------------------------------------------------------------
+
+    // any helper functions should go here or else delete this section
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    return {
+      CallExpression: (node) => {
+        const {
+          callee: { name },
+        } = node;
+        // TODO: look for imported toggles from opticks only
```

Comment:
I do think this one is worthwhile to tackle if we want to publish the plugin publicly. In the JSCodeShift project we do this by looking at named imports from a configurable package name - perhaps a similar approach can be considered here. Let me know if you want to dive into it together.

Reply:
I've implemented the following solution. Let me know if you see any issues with it. https://github.com/viodotcom/opticks/blob/f7e2c79c0f2986130d9320da46d16728f9190314/packages/eslint-plugin/lib/rules/toggle.js#L47-L57

Reply:
> I've implemented the following solution. Let me know if you see any issues with it.

It would work for our use cases, but we also allow imports of specific paths, such as `import {toggle} from 'opticks/lib/optimizely'` - if we still want to allow that, maybe use a substring check with `node.source.value.includes` ?